symbolic-base/src/ZkFold/Algebra/Class.hs

```diff
@@ -365,76 +363,75 @@ class (Ring a, SemiEuclidean a) => Euclidean a where
 
-{- | Class of fields. As a ring, each field is commutative, that is:
-
-[Commutativity] @x * y == y * x@
-
-While exponentiation by an integer is specified in a constraint, a default
-implementation is provided as an @'intPowF'@ function. You can provide a faster
-alternative yourself, but do not forget to check that your implementation
-computes the same results on all inputs.
--}
+-- | Class of fields. As a ring, each field is commutative, that is:
+--
+-- [Commutativity] @x * y == y * x@
+--
+-- While exponentiation by an integer is specified in a constraint, a default
+-- implementation is provided as an @'intPowF'@ function. You can provide a faster
+-- alternative yourself, but do not forget to check that your implementation
+-- computes the same results on all inputs.
 class (Ring a, Exponent a Integer, Eq a) => Field a where
-    {-# MINIMAL (finv | (//)) #-}
-
-    -- | Division in a field. The following should hold:
-    --
-    -- [Division] If @x /= 0@, @x // x == one@
-    -- [Div by 0] @x // zero == zero@
-    -- [Agreement] @x // y == x * finv y@
-    (//) :: a -> a -> a
-    x // y = x * finv y
-
-    -- | Inverse in a field. The following should hold:
-    --
-    -- [Inverse] If @x /= 0@, @x * inverse x == one@
-    -- [Inv of 0] @inverse zero == zero@
-    -- [Agreement] @finv x == one // x@
-    finv :: a -> a
-    finv x = one // x
-
-    -- | @rootOfUnity n@ is an element of a characteristic @2^n@, that is,
-    --
-    -- [Root of 0] @rootOfUnity 0 == Just one@
-    -- [Root property] If @rootOfUnity n == Just x@, @x ^ (2 ^ n) == one@
-    -- [Smallest root] If @rootOfUnity n == Just x@ and @m < n@, @x ^ (2 ^ m) /= one@
-    -- [All roots] If @rootOfUnity n == Just x@ and @m < n@, @rootOfUnity m /= Nothing@
-    rootOfUnity :: Natural -> Maybe a
-    rootOfUnity 0 = Just one
-    rootOfUnity _ = Nothing
+  {-# MINIMAL (finv | (//)) #-}
+
+  -- | Division in a field. The following should hold:
+  --
+  --   [Division] If @x /= 0@, @x // x == one@
+  --   [Div by 0] @x // zero == zero@
+  --   [Agreement] @x // y == x * finv y@
+  (//) :: a -> a -> a
+  x // y = x * finv y
+
+  -- | Inverse in a field. The following should hold:
+  --
+  --   [Inverse] If @x /= 0@, @x * inverse x == one@
+  --   [Inv of 0] @inverse zero == zero@
+  --   [Agreement] @finv x == one // x@
+  finv :: a -> a
+  finv x = one // x
+
+  -- | @rootOfUnity n@ is an element of a characteristic @2^n@, that is,
+  --
+  --   [Root of 0] @rootOfUnity 0 == Just one@
+  --   [Root property] If @rootOfUnity n == Just x@, @x ^ (2 ^ n) == one@
+  --   [Smallest root] If @rootOfUnity n == Just x@ and @m < n@, @x ^ (2 ^ m) /= one@
+  --   [All roots] If @rootOfUnity n == Just x@ and @m < n@, @rootOfUnity m /= Nothing@
+  rootOfUnity :: Natural -> Maybe a
+  rootOfUnity 0 = Just one
+  rootOfUnity _ = Nothing
 
 intPowF :: Field a => a -> Integer -> a
+
 -- | A default implementation for integer exponentiation. Uses only natural
```

Comment:
Wrong type of comment use here, either `-- ^` needed here or type declaration should go below the doc comment